### PR TITLE
Improve port library startup failure return codes

### PR DESCRIPTION
**Pull request by pshipton**
Issue https://github.com/eclipse-openj9/openj9/issues/12526

### CODE REVIEW

**Comment by pshipton** [comment]
@keithc-ca can you review pls.

File: include_core/omrporterror.h [hunk 1]
```diff
@@ -104,7 +104,18 @@
 #define OMRPORT_ERROR_STARTUP_SHMEM (OMRPORT_ERROR_STARTUP_BASE -24)
 #define OMRPORT_ERROR_STARTUP_SHMEM_MOSERVICE (OMRPORT_ERROR_STARTUP_BASE -25)
 #define OMRPORT_ERROR_STARTUP_SYSINFO_MONITOR_INIT (OMRPORT_ERROR_STARTUP_BASE -26)
-#define OMRPORT_ERROR_STARTUP_SIGNAL_TOOLS (OMRPORT_ERROR_STARTUP_BASE -27)
+#define OMRPORT_ERROR_STARTUP_SIGNAL_TOOLS1 (OMRPORT_ERROR_STARTUP_BASE -27)
```
Comment (keithc-ca) [inline include_core/omrporterror.h:107]: This has the potential to break downstream projects (by removing `OMRPORT_ERROR_STARTUP_SIGNAL_TOOLS`).
I suggest we keep it (perhaps as an alias for `OMRPORT_ERROR_STARTUP_SIGNAL_TOOLS1`) which would also eliminate the need to change the Windows implementations of `omrsignal.c`.

Comment (pshipton) [reply include_core/omrporterror.h:107]: I only added OMRPORT_ERROR_STARTUP_SIGNAL_TOOLS  two days ago in https://github.com/eclipse/omr/pull/6231

Comment (pshipton) [reply include_core/omrporterror.h]: Updated.

File: include_core/omrporterror.h [hunk 2]
```diff
@@ -105,6 +105,18 @@
 #define OMRPORT_ERROR_STARTUP_SHMEM_MOSERVICE (OMRPORT_ERROR_STARTUP_BASE -25)
 #define OMRPORT_ERROR_STARTUP_SYSINFO_MONITOR_INIT (OMRPORT_ERROR_STARTUP_BASE -26)
 #define OMRPORT_ERROR_STARTUP_SIGNAL_TOOLS (OMRPORT_ERROR_STARTUP_BASE -27)
+#define OMRPORT_ERROR_STARTUP_SIGNAL_TOOLS1 (OMRPORT_ERROR_STARTUP_BASE -27) /* duplicate of OMRPORT_ERROR_STARTUP_SIGNAL_TOOLS */
```
Comment (keithc-ca) [inline include_core/omrporterror.h:108]:
```
#define OMRPORT_ERROR_STARTUP_SIGNAL_TOOLS1 OMRPORT_ERROR_STARTUP_SIGNAL_TOOLS
```
would clearly indicate it's a duplicate

Comment (pshipton) [reply include_core/omrporterror.h]: Yes that's better. Updated.

**Review by keithc-ca — APPROVED** [review]

**Comment by pshipton** [comment]
@0xdaryl can you pls have a look.

**Comment by 0xdaryl** [comment]
Jenkins build all

**Review by 0xdaryl — APPROVED** [review]

**Pull request merged**